src/operator/gitea.ts

```diff
@@ -0,0 +1,161 @@
+      const execCommand = [
+        'sh',
+        '-c',
+        `gitea admin auth update-oauth --id 1 --group-team-map '${buildTeamString(teamNamespaces)}'`,
+      ]
+      if (object && object.metadata && object.metadata.namespace && object.metadata.name) {
+        const exec = new k8s.Exec(kc)
+        // Run gitea CLI command to update the gitea oauth group mapping
+        await exec
+          .exec(
+            object.metadata?.namespace,
+            object.metadata?.name,
+            'gitea',
+            execCommand,
+            process.stdout as stream.Writable,
+            process.stderr as stream.Writable,
+            process.stdin as stream.Readable,
+            false,
+            (status: k8s.V1Status) => {
+              console.log('Exited with status:')
+              console.log(JSON.stringify(status, null, 2))
+            },
+          )
+          .catch((error) => {
+            console.debug('Error occurred during exec:', error)
+          }) // needs to be done better, currently always fires
+          .then(() => console.debug('Commands are executed!'))
+      }
+    } else {
+      console.debug('No team namespaces found')
+    }
+  } catch (error) {
+    console.debug(
+      `Error updating IDP group mapping: statuscode: ${error.response.body.code} - message: ${error.response.body.message}`,
+    )
+  }
+}
+
+async function checkGiteaContainer(): Promise<ConditionCheckResult> {
+  let giteaPod: any
+  try {
+    giteaPod = (await k8sApi.readNamespacedPod('gitea-0', 'gitea')).body
+  } catch (error) {
+    console.debug('Gitea pod could not be found')
+    return { ready: false, pod: undefined }
+  }
+  // Check if 'gitea-0' pod has a container named 'gitea'
+  const containerStatuses = giteaPod.status?.containerStatuses || []
+  const giteaContainer = containerStatuses.find((container) => container.name === 'gitea')
+  // Check if the gitea container exists'
+  if (giteaContainer === undefined) {
+    console.debug('Gitea container is not found')
+    return { ready: false, pod: undefined }
+  }
+  // Check if the gitea container has status 'READY'
+  if (!giteaContainer?.ready) {
+    console.debug('Gitea container is not ready: ', giteaContainer.state!)
+    return { ready: false, pod: giteaPod }
+  }
+  return { ready: true, pod: giteaPod }
+}
+
+export default class MyOperator extends Operator {
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  protected async init() {
+    // Watch all namespaces
+    try {
+      await this.watchResource('', 'v1', 'namespaces', async (e) => {
+        const { object }: { object: k8s.V1Pod } = e
+        const { metadata } = object
+        // Check if namespace starts with prefix 'team-'
+        if (metadata && !metadata.name?.startsWith('team-')) return
+        if (metadata && metadata.name === 'team-admin') return
+        let giteaPod = await checkGiteaContainer()
+        while (!giteaPod.ready) {
```

Comment:
I understand that gitea Pod must be ready, so CLI commands can be performed. 
However I would extend error handling to broader scope. 
The operator should retry in case of any failure. 
Therefore you may consider using try catch for `execGiteaCLICommand` and in case of failure use retry it after 30 seconds. It would not only simply the code but also enhance it.

Reply:
This would simplify the code yes, but I also like that we can give a more precise error at the moment regarding the status of the gitea pod. Which would be harder to do if we simplify this.